### ISSUE TITLE
P1: Reconnect CFD as multi-fidelity oracle, surrogate target, and generation-time critic

### ISSUE THREAD

**Issue by iamdarshg** [iamdarshg/research-paper#15]
## Severity
P1 / High

## Problem
The CFD solver is present and improved, but it is not yet used as a full design oracle for mission-compliant generation. At the moment, CFD is used inside training through `AerodynamicLoss` on occasional batches and during evaluation, while external validation is gated by `validation_probability` and defaults to disabled.

This creates three gaps:
1. CFD results are not systematically stored as reusable labels.
2. The diffusion model has no differentiable aerodynamic surrogate to guide training continuously.
3. Generation does not rank/filter candidates using a structured multi-fidelity CFD pipeline.

## Why this matters
For full aircraft-body generation, a single sampled geometry should not be trusted. The system needs a loop that samples many candidates, applies constraints, scores them quickly, validates top candidates with D3Q27, and optionally promotes external PDE/OpenFOAM results as higher-quality labels.

## Proposed fix
Use CFD in three independent roles:

### 1. Offline label generator
For generated or dataset geometries, store:
- geometry
- mission profile
- constraint profile
- Cd / Cl / Cm where available
- pressure and velocity fields
- label tier: `lbm_raw`, `lbm_calibrated`, `external_pde`
- convergence and force-stability metadata

This can extend the existing `GroundTruthExporter` rather than replacing it.

### 2. Differentiable surrogate target
Train an `AeroSurrogate` that predicts aerodynamic metrics from geometry + mission profile:

```python
AeroSurrogate(voxel_channels, mission_embedding) -> {
    "Cd": ..., 
    "Cl": ..., 
    "Cm": ...,
    "convergence_score": ...,
    "separation_risk": ...
}
```

Use D3Q27 labels for broad coverage and external PDE/OpenFOAM labels for correction/calibration where available.

### 3. Generation-time critic
Update generation to sample multiple candidates, rank/filter them, and validate top candidates:

```python
candidates = generator.sample_many(mission, n=64)
candidates = constraint_projector.batch_project(candidates)
scores = aero_surrogate.rank(candidates, mission)
top = select_top_k(candidates, scores, k=8)
cfd_results = d3q27.evaluate_batch(top)
best = choose_feasible(cfd_results, mission_constraints)
```

## Acceptance criteria
- A reusable CFD label dataset is created with label source/tier metadata.
- A lightweight `AeroSurrogate` can be trained and queried during generation.
- `generate` or a new `generate_candidates` command can sample multiple candidates and rank them.
- D3Q27 is run on top candidates before export.
- External PDE/OpenFOAM validation can be enabled for final candidates without changing the training loop.
- Tests verify label-tier handling and candidate ranking behavior.

## Independent scope
This can be implemented after or alongside mission conditioning. It does not require the full manufacturing/mechanical constraint system, only geometry tensors and mission metadata inputs.
